dist/clouduploader.rb

```diff
@@ -5,98 +5,83 @@
+        'ec2uploadimg',
+        "--description='obs uploader'",
+        '--machine=x86_64',
+        "--name=#{data['ami_name']}",
+        "--region=#{data['region']}",
+        "--secret-key=#{credentials.secret_access_key}",
+        "--access-id=#{credentials.access_key_id}",
+        "--session-token=#{credentials.session_token}",
+        '--verbose',
+        image
+      ) do |_stdin, stdout_stderr, _wait_thr|
+        while line = stdout_stderr.gets
+          STDOUT.write(line)
+        end
+        status = wait_thr.value
+        abort unless status.success?
+      end
+    end
   end
-end
 
-def upload_image_to_ec2(image, data, result_path)
-  STDOUT.write("Start uploading image #{image}.\n")
+  def self.upload(platform, image_path, job_data_file, image_filename)
+    STDOUT.sync = true
+    start = Time.now
 
-  credentials = get_ec2_credentials(data)
-  command = [
-    'ec2uploadimg',
-    "--description='obs uploader'",
-    '--machine=x86_64',
-    "--name=#{data['ami_name']}",
-    "--region=#{data['region']}",
-    "--secret-key=#{credentials.secret_access_key}",
-    "--access-id=#{credentials.access_key_id}",
-    "--session-token=#{credentials.session_token}",
-    '--verbose'
-  ]
+    job_data = JSON.parse(File.read(job_data_file))
+    FileUtils.ln_s(image_path, File.join(Dir.pwd, image_filename)) # link file into working directory
 
-  if data['vpc_subnet_id']
-    command << "--vpc-subnet-id=#{data['vpc_subnet_id']}"
-  end
-  command << image
-
-  Open3.popen2e(*command) do |_stdin, stdout_stderr, wait_thr|
-    Signal.trap("TERM") {
-      # We just omit the SIGTERM because otherwise we would not get logs from ec2uploadimg
-      STDOUT.write("Received abort signal, waiting for ec2uploadimg to properly clean up.\n")
-    }
-    while line = stdout_stderr.gets
-      STDOUT.write(line)
-      write_result($1, result_path) if line =~ /^Created\simage:\s+(ami-[\w]+)$/
+    case platform
+    when 'ec2'
+      THIRTY_MINUTES = 1800
+      ENV['PYTHONUNBUFFERED'] = '1'
+      CloudUploader::EC2.upload(image_filename, job_data)
+    when 'azure'
+      # TODO: Add the Azure upload code here
+    else
+      abort('No valid platform. Valid platforms are ec2 and azure.')
     end
-    status = wait_thr.value
-    abort unless status.success?
-  end
-end
 
-def write_result(result, result_path)
-  File.open(result_path, "w+") { |file| file.write(result) }
-end
-
-if platform == 'ec2'
-  upload_image_to_ec2(filename, data, result_path)
-else
-  abort('No valid platform. Valid platforms is ec2.')
+    diff = Time.now - start
+    STDOUT.write("Upload took: #{Time.at(diff).utc.strftime("%H:%M:%S")}\n")
+  end
 end
 
-diff = Time.now - start
-STDOUT.write("Upload took: #{Time.at(diff).utc.strftime("%H:%M:%S")}")
+HOME = '/etc/obs/cloudupload'.freeze
+ENV['HOME'] = HOME
+raise 'Wrong number of arguments, please provide: user platform upload_file targetdata filename' unless ARGV.length == 5
+CloudUploader.upload(*ARGV)
```

Comment:
The ` src/backend/build ` code below doesn't belong here :-)

Reply:
No idea how this got in... neither how to get rid of it :smile:

Reply:
@mdeniz I can't tell how it got there, but this is how you can get rid of it :smile:

```bash
git rebase -i $commit_that_added_it
chose to edit that commit
# Reset only the changes of ../backend/build
git reset $the_commit_that ../backend/build
# Ammend the reverted changes to the commit
git commit --amend
# Clear your working dir
git checkout ../backend/build
git rebase --continue
```

Reply:
Thanks